src/LibraryManager/Providers/Cdnjs/CdnjsProvider.cs

```diff
@@ -235,17 +235,17 @@ public ILibraryCatalog GetCatalog()
             return null;
         }
 
-        private string GetLibraryName(ILibraryInstallationState state)
+        public string GetLibraryName(string libraryId)
         {
-            string[] args = state.LibraryId.Split('@');
+            string[] args = libraryId.Split('@');
             string name = args[0];
 
             return name;
         }
 
-        private string GetLibraryVersion(ILibraryInstallationState state)
+        public string GetLibraryVersion(string libraryId)
         {
-            string[] args = state.LibraryId.Split('@');
+            string[] args = libraryId.Split('@');
             string version = args[1];
```

Comment:
>string version = args[1]; [](start = 12, length = 25)

what happens if no '@' ?

Reply:
Planning on fixing this as part of #123. Unless, we get approval for the proper fix to make to preview3 escrow. 
@jodavis , can you please comment?

Reply:
The goal for Preview 3 is to just get us to not crash. As long as the IndexOutOfRange exception is handled, I think we're okay for now. For Preview 4, we should handle this error case better.

---
In reply to: [194856346](https://github.com/aspnet/LibraryManager/pull/115#discussion_r194856346) [](ancestors = 194856346)